src/main/java/org/checkstyle/autofix/CheckstyleAutoFix.java

```diff
@@ -49,12 +75,69 @@ public String getViolationReportPath() {
         return violationReportPath;
     }
 
+    public String getCheckstyleConfigurationPath() {
+        return checkstyleConfigurationPath;
+    }
+
+    public String getPropertiesPath() {
+        return propertiesPath;
+    }
+
     @Override
     public List<Recipe> getRecipeList() {
+        try {
+            final Configuration config = loadCheckstyleConfiguration();
+
+            final List<CheckstyleViolation> violations = CheckstyleReportParser
+                    .parse(Path.of(getViolationReportPath()));
 
-        final List<CheckstyleViolation> violations = CheckstyleReportParser
-                .parse(Path.of(getViolationReportPath()));
+            return CheckstyleRecipeRegistry.getRecipes(violations, config);
 
-        return CheckstyleRecipeRegistry.getRecipes(violations);
+        }
+        catch (CheckstyleException | IOException exception) {
+            throw new IllegalArgumentException("Failed to load Checkstyle"
+                    + " configuration or parse violations", exception);
+        }
     }
+
+    private Configuration loadCheckstyleConfiguration() throws CheckstyleException, IOException {
```

Comment:
Could you check Checkstyle code please? Can we reuse its logic to load config given a config path uri, so we don't need to handle it ourselves like you did it in this method?

Reply:
This will be done in a separate PR